BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/MainActivity.java

```diff
@@ -26,95 +28,159 @@
                 }
                 SharedPreferences sharedPref = getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
-                sharedPref.edit().putString(getString(R.string.preference_login_token), response.body().getToken()).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_token), response.headers().get("access-token")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_token_type), response.headers().get("token-type")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_client), response.headers().get("client")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_expiry), response.headers().get("expiry")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_uid), response.headers().get("uid")).commit();
                 navigateToHome();
             }
 
             @Override
-            public void onFailure(Call<BeeTrackerCaller.SignupResponse> call, Throwable t) {
-                Log.e(TAG, "There was an error with the loginCallback + " + t.toString());
+            public void onFailure(Call<BeeTrackerCaller.EmailPasswordSigninResponse> call, Throwable t) {
+                Log.e(TAG, "There was an error with the login callback + " + t.toString());
                 t.printStackTrace();
                 LoginManager.getInstance().logOut();
                 spinningIconDialog.dismiss();
                 showErrorDialog(getString(R.string.error_message_login));
             }
         };
-    }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        callbackManager.onActivityResult(requestCode, resultCode, data);
+        TextView signUpButton = (TextView) findViewById(R.id.signup_with_email);
+        signUpButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                spinningIconDialog = new SpinningIconDialog();
+                spinningIconDialog.show(getFragmentManager(), "SpinningPopup");
+
+                BeeTrackerCaller caller = new BeeTrackerCaller();
+                try {
+                    String email = ((EditText) findViewById(R.id.email_input)).getText().toString();
+                    String password = ((EditText) findViewById(R.id.password_input)).getText().toString();
+                    Call<BeeTrackerCaller.EmailPasswordSignupResponse> call = caller.emailPasswordSignup(email, password);
+                    call.enqueue(emailPasswordSignupCallback);
+
+                    SharedPreferences sharedPref = getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+                    sharedPref.edit().putString(getString(R.string.signup_email), email).commit();
```

Comment:
Why is this being stored?

Reply:
The API doc says "This confirms the email. Next we login. This does not necessarily have to be two steps to the user; we can send the API request to /sign_in without asking the user for password again if we have stored it locally".

I decided to store the email/password that the user uses to sign up so we can do the one-step signup -> login, but I'm not sure how the app is supposed to know that a user has confirmed their email. Or maybe Dinah meant the email and password (in secure text mode) would be displayed on the login screen for convenience after they re-open the app after confirming their email, but they still have to press the log in button. Maybe @dinahshi can help clarify?

Reply:
I was thinking what Steven had in mind: one step signup so the user doesn't have to enter their email or click login. This is where that confirm_success_url would come in; if you can launch the app and trigger the automatic login, it'd be guaranteed to work since the user has just confirmed their email. Or the app could put a login button on the "waiting for confirmation" modal that the user can click once the app is launched (and do nothing if login fails).

We're still figuring out the ux for this so we'll keep you posted. Important thing is if we are storing the password no the device, it should be stored securely.